### ISSUE TITLE
Rebuild Library command

### ISSUE THREAD

**Issue by Magic815** [scakemyer/plugin.video.quasar#107]
So this was born out of a discussion between @i96751414 and myself, but I figured it could use it's own issue submission for traceability.

Quasar will soon hold quite a long list of added features thanks to the tireless effort of the devs. However, I'm realizing that with these added features, we'll likely want to have a 'rebuild library' command that can be run from the settings.

Just a few current feature ideas that could benefit: Toggle Specials with TV Shows, Trakt Integration, Duplicate Handling. If those all got implemented, we could have scenarios such as the following:

- The user didn't want specials to display for TV Shows, but now suddenly does. A "rebuild library" would be good for forcing all of the current shows to add specials to their episode list. In reverse, someone toggling off specials would want them deleted from all TV episode lists.
- With Trakt, I could have a movie on my watchlist that I finally get around to watching, so then it gets removed from the watchlist on Trakt. If I'm currently integrating my watchlist into my Movies library via Quasar, I'd then want Quasar to delete the .strm it had created for that movie. Same thing if I remove any TV show or movie from any Trakt list. It'd be nice for a 'rebuild' to allow Quasar to clean up those .strms that I'd no longer want.
- With duplicate handling, I might have had a movie on my Trakt list that I also owned and had a local copy of on my computer. So Quasar would have ignored the integrating of a .strm file for that movie from Trakt. However, if I later delete the local copy of the movie to save space on my computer, I'd then want the 'rebuild' command to then integrate a .strm for that movie (as it's no longer a duplicate).

Put simply, it would almost be a force sync of the library for any updates that occurred in Trakt or in Quasar's addons settings. However, I imagine the rebuild could be slow, so we wouldn't want this to be something that happened at startup everytime, but just something that could be force through the addon settings. Thoughts? Comments?

**Comment by scakemyer** [comment]
Quoting myself from #104 (should've just answered here but hey..)
> A rebuild function would be great, and could be called automatically when settings are changed, just like the libtorrent reload of settings, while also allowing the user to trigger the action manually. Since it'd be an expensive thing, it wouldn't be called on library updates, and should probably show a background progress dialog.

**Comment by i96751414** [comment]
Umm, that's nice. But can it be called when a specific setting is changed?

**Comment by hfmls** [comment]
I found myself using "collection list" instead of watchlist, i don't think watchlist works that well with trakt, sometimes tv shows just vanish from my "favourites" and i forget all about that tv show.

Using Collection list from trakt i get all tv shows i want and scrobble them without any problems. I add,remove, etc etc without any problems.
I used this in GENISIS, we could chose between watchlist or collection list.

**Comment by nix87** [comment]
Will this feature work as subscription feature in Istream? E.g. in a moment when you added specific tv show that tv show have 10 episodes but after week normally it will have 11 but in your library is same 10 episodes. You use rebuild library command and update that tv show?

**Comment by i96751414** [comment]
@nix87 that is already implemented. Whenever you start Kodi (and every 8 hours with Kodi running) the Library will update itself automatically

**Comment by nix87** [comment]
oh sorry guys for my ignorance then, and thank you

**Comment by hfmls** [comment]
so this is like going into "tv shows" press left, and update collection?

is it the same behaviour?

**Comment by dkrushev** [comment]
Updating the library does not work for me (0.9.66). I'm going to settings of the add-on and updating the library, then go to the TV Show section of Kodi and update library from there as well, but new episodes do not appear. Going through Quasar -> TV Shows -> Search for TV Show -> opening last season does not show latest episode as well (it has been 4 days since the episode was released). Changing the database to Trakt and back to TVDB does not help as well.

**Comment by scakemyer** [comment]
You probably have cache files from before the caching refactor, which had way too long expiration values. You can try the "Clear cache" option from the Advanced settings, but since it was being overused I had to limit its destructive power to lower the number of requests to TMDB. So, I won't recommend this often, but in your case, try going in your [userdata](http://kodi.wiki/view/Userdata) folder, and remove the files in Quasar's cache folder. Restart Kodi, wait a few minutes to let Quasar rebuild its cache, then go to your show's season and everything should be there. Note that you most probably won't ever have to do this again; seasons now have cache expiration times of 15 minutes, so new episodes would appear as soon as they're available.

**Comment by joaopauloss** [comment]
> Updating the library does not work for me (0.9.66). I'm going to settings of the add-on and updating the library, then go to the TV Show section of Kodi and update library from there as well, but new episodes do not appear. Going through Quasar -> TV Shows -> Search for TV Show -> opening last season does not show latest episode as well (it has been 4 days since the episode was released). Changing the database to Trakt and back to TVDB does not help as well.

Same problema here,

> So, I won't recommend this often, but in your case, try going in your userdata folder, and remove the files in Quasar's cache folder. Restart Kodi, wait a few minutes to let Quasar rebuild its cache, then go to your show's season and everything should be there.

Tried to do what @scakemyer said, but didn't work too. Any ideas ?

Thanks.

**Comment by nix87** [comment]
It's working perfectlly for me on Windows.

No problems with tv shows updated automatically.

**Comment by scakemyer** [comment]
@joaopauloss I think you're also running into #750
